From fcefdb5c18e3925f99320b051af28c34f095fb83 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Vojt=C4=9Bch=20Hejsek?= <hejsekvojtech@gmail.com>
Date: Thu, 8 Jun 2017 17:39:37 +0200
Subject: [PATCH] prepareWithoutScan() on MediaTek

---
 media/java/android/media/MediaPlayer.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/media/java/android/media/MediaPlayer.java b/media/java/android/media/MediaPlayer.java
index b3f25eea565..1c78f758215 100644
--- a/media/java/android/media/MediaPlayer.java
+++ b/media/java/android/media/MediaPlayer.java
@@ -1172,6 +1172,10 @@ public void prepare() throws IOException, IllegalStateException {
      * @throws IllegalStateException if it is called in an invalid state
      */
     public native void prepareAsync() throws IllegalStateException;
+    
+    public void prepareWithoutScan() throws IOException, IllegalStateException {
+        _prepare();
+    }
 
     /**
      * Starts or resumes playback. If playback had previously been paused,
